Bottom: 27e4b4168886e7dc0b37720773a0cd1c06cd9d9f
Top:    eccb4f98f47fb300ddcbc7a91bf473096d26a08b
Author: Jon Smirl <jonsmirl@gmail.com>
Date:   2014-07-16 19:14:21 -0400

em patch for mono


---

diff --git a/sound/soc/sunxi/sunxi-codec.c b/sound/soc/sunxi/sunxi-codec.c
index 3941a0b..effe0c8 100644
--- a/sound/soc/sunxi/sunxi-codec.c
+++ b/sound/soc/sunxi/sunxi-codec.c
@@ -353,14 +353,14 @@ static int sunxi_codec_hw_params(struct snd_pcm_substream *substream,
 	}
 	if (substream->stream == SNDRV_PCM_STREAM_PLAYBACK) {
 		regmap_update_bits(priv->regmap, SUNXI_DAC_FIFOC, 7 << DAC_FS, rate << DAC_FS);
-		if (substream->runtime->channels == 1) {
+		if (params_channels(params) == 1) {
 			regmap_update_bits(priv->regmap, SUNXI_DAC_FIFOC, 1 << DAC_MONO_EN, 1 << DAC_MONO_EN);
 		} else {
 			regmap_update_bits(priv->regmap, SUNXI_DAC_FIFOC, 1 << DAC_MONO_EN, 0 << DAC_MONO_EN);
 		}
 	} else  {
 		regmap_update_bits(priv->regmap, SUNXI_ADC_FIFOC, 7 << 29, rate << 29);
-		if (substream->runtime->channels == 1) {
+		if (params_channels(params) == 1) {
 			regmap_update_bits(priv->regmap, SUNXI_ADC_FIFOC, 1 << ADC_MONO_EN, 1 << ADC_MONO_EN);
 		} else {
 			regmap_update_bits(priv->regmap, SUNXI_ADC_FIFOC, 1 << ADC_MONO_EN, 0 << ADC_MONO_EN);
